lib/network/kmip/codec/ttlv.js

```diff
@@ -0,0 +1,434 @@
+'use strict'; // eslint-disable-line
+/* eslint dot-notation: "off" */
+
+const KMIPTags = require('../tags.json');
+const KMIPMessage = require('../Message.js');
+
+const UINT32_MAX = Math.pow(2, 32);
+
+function _ttlvPadVector(vec) {
+    let length = 0;
+    vec.forEach(buf => {
+        if (!(buf instanceof Buffer)) {
+            throw Error('Not a Buffer');
+        }
+        length += buf.length;
+    });
+    const paddingLength = (Math.ceil(length / 8) * 8) - length;
+    if (paddingLength > 0) {
+        vec.push(new Buffer(paddingLength).fill(0));
+    }
+    return vec;
+}
+
+function _throwError(logger, msg, data) {
+    logger.error(msg, data);
+    throw Error(msg);
+}
+
+function TTLVCodec() {
```

Comment:
curious as to why `class` isn't used instead

Reply:
It simplifies the code and the scoping of the various encoding/decoding structures which are kept as closures in member functions. I started implementing this using the more modern class construct as well, but in the end this version using a function as a constructor is simpler to understand and maintain.

Reply:
Oh my preference still goes to the class construct, I just didn't succeed into articulating a digestible code with a class. IIRC mostly because of the nomenclature which is amended with extensions on a per instance basis (for the purpose of keeping the codec part simple).